### PR TITLE
Fix Redis with unix socket #730

### PR DESCRIPTION
**Pull request by epma01**
Fix Redis with unix socket https://github.com/nextcloud/docker/issues/730

### CODE REVIEW

File: lib/private/RedisFactory.php [hunk 1]
```diff
@@ -72,9 +72,12 @@ private function create() {
 			}
 			if (isset($config['port'])) {
 				$port = $config['port'];
-			} else {
+			} else if ($host[0] != '/') {
 				$port = 6379;
 			}
+			else {
```
Comment (rullzer) [inline lib/private/RedisFactory.php]: Could you change this to

`} else {`

**Review by rullzer — CHANGES_REQUESTED** [review]

File: lib/private/RedisFactory.php (hunk 1 above)
Comment (epma01) [reply lib/private/RedisFactory.php]: done

**Review by rullzer — APPROVED** [review]
If you could squash your commits that would be even better :)
But thnx for the patch

File: lib/private/RedisFactory.php [hunk 2]
```diff
@@ -72,8 +72,10 @@ private function create() {
 			}
 			if (isset($config['port'])) {
 				$port = $config['port'];
-			} else {
+			} else if ($host[0] != '/') {
```
Comment (kesselb) [inline lib/private/RedisFactory.php]: Does it work if you use `!==` here? It's a more strict comparison.

Comment (MorrisJobke) [reply lib/private/RedisFactory.php]: That would be indeed better.

Comment (epma01) [reply lib/private/RedisFactory.php]: strict comparison works. changed PR.

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**